peas/weather_web_data.py

```diff
@@ -0,0 +1,188 @@
+        except AttributeError:
+            cache_age = self.web_data.get('cache_age', 1.382e10) * u.year
+
+        if cache_age > self.max_age:
+            # Download met data file
+            m = requests.get(self.web_data.get('link'))
+            # Remove the annoying " and newline between the date and time
+            met = m.text.replace('."\n',' ')
+            # Remove the " before the date
+            met = met.replace('" ', '')
+
+            # Parse the tab delimited met data into a Table
+            t = Table.read(met, format='ascii.no_header', delimiter='\t',
+                                names=self.web_data.get('column_names'))
+
+            # Convert time strings to Time
+            t['Time (UTC)'] = Time(t['Time (UTC)'], format='mixed_up_time')
+            # Change string format to ISO
+            t['Time (UTC)'].format = 'iso'
+            # Convert from AAT standard time to UTC
+            t['Time (UTC)'] = t['Time (UTC)'] - 10 * u.hour
+
+            col_names = self.web_data.get('column_names')
+            col_units = self.web_data.get('column_units')
+
+            if len(col_names) != len(col_units):
+                self.logger.debug('Number of columns does not match number of units given')
+
+            # Set units for items that have them
+            for i in range(1, len('col_names')):
+                t[col_names[i]].unit = col_units[i]
+
+            self._met_data = t
+
+        return self._met_data
+
+    def _get_cloud_safety(self, current_values):
+        entries = self.weather_entries
+
+        sky_diff = weather_entries[self.web_data.get('sky_ambient')]
+        sky_diff_u = weather_entries[self.web_data.get('sky_ambient_uncertainty')]
+
+        max_sky_diff = sky_diff + sky_diff_u
+        last_cloud = sky_diff
+
+        super._get_cloud_safety()
+
+        return cloud_condition, sky_safe
+
+    def _get_wind_safety(self, current_values):
+        entries = self.weather_entries
+
+        # Wind (average and gusts)
+        wind_speed = entries[self.web_data.get('average_wind_speed')]
+        wind_gust = entries[self.web_data.get('maximum_wind_gust')]
+
+        super._get_wind_safety()
+
+        return (wind_condition, wind_safe), (gust_condition, gust_safe)
+
+    def _get_rain_alarm_safety(self, current_values):
+        super._get_rain_safety()
+        threshold_rain = self.web_data.get('threshold_rain', 0)
+        threshold_wet = self.web_data.get('threshold_wet', 0)
+
+        # Rain
+        rain_sensor = entries[self.web_data('rain_sensor')]
+        rain_flag = entries[self.web_data('rain_flag')]
+        wet_flag = entries[self.web_data('wet_flag')]
+
+        if len(rain_sensor) == 0 and len(rain_flag) == 0 and len(wet_flag) == 0:
+            rain_safe = False
+            rain_condition = 'Unknown'
+        else:
+            # Check current values
+            if rain_sensor > threshold_rain and rain_flag > threshold_rain:

```

Comment:
Pretty sure we want that to be `or` instead of `and`.  If either of the two rain sensors in the AAT weather station say it's raining, we should assume it's raining.

Reply:
Has been changed.